Remotion/Web/Core/UI/Controls/CommandInfo.cs

```diff
@@ -157,7 +157,22 @@ private void AddDiagnosticMetadataAttributes (HtmlTextWriter writer)
 
     private bool IsTriggeringNavigation ()
     {
-      return _href != "#";
+      if (string.IsNullOrEmpty(_onClick))
+      {
+        return true;
+      }
+
+      if (_onClick.Contains("__doPostBack"))
+      {
+        return false;
+      }
+
+      if (_href == null || _href.StartsWith("#"))
+      {
+        return false;
+      }
+
+      return true;
```

Comment:
stylistic remarks:
* no braces in code structure like this (just simple return statements. it makes the code compacter without trading readability for it.
* this sort of logic isn't really a set of early exit conditions but a group of connected conditions that work better as an if/elseif/else. Can't promise there's not other places in re-motion with this semantics and style, but it's a bit clearer if done as a single block.

actual logic still has the bug I was looking to fix. The test for javascript is necessary but not sufficient when it comes to determining whether this is a navigation-triggering anchor. If the anchor does not have an HREF or the HREF starts with a "#", it will still not trigger a navigation. This means returning "true" if there's no javascript covers only part of the required conditions. You need to go with something like the following, starting with the most dominant conditions first.

```suggestion
      if (_href == null) // without an href, an anchor cannot do a navigation
        return false;
      else if (_href.StartsWith("#")) // when the href points to a page-local target, it's not considered a navigation
        return false;
      else if (_onClick == null) // if there is no onclick-handler and we have already confirmed that there is a href, it can only be a navigation. I dropped the empty-check since we only need to guard the null-reference and it makes for better symmetry.
        return true;
      else if (_onClick.Contains("__doPostBack")) // this could be here, it could also be at the beginning, if the null-check is part of the same condition.
        return false;
      else
        return true;
```

Please remove the comments, they're just instructional.

alternative:
```
      else if (_onClick?.Contains("__doPostBack"))
        return false;
      if (_href == null)
        return false;
      else if (_href.StartsWith("#"))
        return false;
      else
        return true;
```
This version would also work, but it has two conditions in the same IF. Given we have nullable reference types, this should be safe from null-reference execptions, but we still want to ensure that we have a test for both the null-value and the __doPostBack cases. When tracking code coverage, the first variant is easier to verify.